Admin panel

alekseirechitskiy opened the pull request:
Created Admin Panel

alekseirechitskiy committed:
initial commit

alekseirechitskiy committed:
added layout of LogIn page;
added MUI lib;
added font Roboto;
added some scss-variables for admin panel

alekseirechitskiy committed:
added AdminHeader

alekseirechitskiy committed:
added layout of the CreatePoint page

alekseirechitskiy committed:
fixed layout of the CreatePoint page

alekseirechitskiy committed:
added layout of the EditPoint page without map

alekseirechitskiy committed:
fixed the container adaptability

alekseirechitskiy committed:
added editconfig

alekseirechitskiy committed:
fixed logo in adminHeader

alekseirechitskiy committed:
added adaptability of titles and checkboxex

alekseirechitskiy committed:
added map to the EditPoint page

alekseirechitskiy committed:
added layout of the Point page

alekseirechitskiy committed:
added ManagePoints page

alekseirechitskiy committed:
added PointsMap tab

alekseirechitskiy committed:
added PointsList tab with raw table layout

alekseirechitskiy committed:
added AdminHeader to every page

alekseirechitskiy committed:
deleted unnecessary code

alekseirechitskiy committed:
added temporary component PointsListPag with raw pagination

alekseirechitskiy committed:
added form validation for Login page

alekseirechitskiy committed:
added filters functionality to PointsMap tab

alekseirechitskiy committed:
removed temporary component PointsListPag

alekseirechitskiy committed:
made AdminHeader fixed;
implemented field validation (organizationName) in CreatePoint component

alekseirechitskiy committed:
added validation to the whole form in the CreatePoint component

alekseirechitskiy committed:
added point data from MapState to the Point component for data rendering

alekseirechitskiy committed:
added PointForm component, decomposed CreatePoint and EditPoint components, cleared the code

alekseirechitskiy committed:
added AdminPanel page

alekseirechitskiy committed:
added regEexp for coordinates

alekseirechitskiy committed:
added custom function 'toBoolean'

alekseirechitskiy committed:
added styles to the tablePagination in the PointsList component

alekseirechitskiy committed:
fixed some semantic errors in MUI, cleaned up the code

alekseirechitskiy committed:
cleaned up the code

alekseirechitskiy committed:
.editorconfig and .prettierrc files in .gitignore file

alekseirechitskiy committed:
installed react-router-dom

alekseirechitskiy committed:
added routes to the AdmiPanel

alekseirechitskiy committed:
added to the submit-button dynamically change of the attribute 'disable'

alekseirechitskiy committed:
added functionality to the Cancel button in the PointForm component

alekseirechitskiy committed:
added temporary state functionality for points

alekseirechitskiy committed:
cleaned up the code

alekseirechitskiy committed:
changed paths to the admin's pages

alekseirechitskiy committed:
fixed hovers in Point and scrolls in PointsList

alekseirechitskiy committed:
fixed outlines, link in Login, pencil button

Valeria310 reviewed (approved):
Хорошая работа, молодец!

fstk1337 commented on .editorconfig:
Предлагаю убрать файл из репозитория.

alekseirechitskiy replied:
Удалил.

fstk1337 commented on .prettierrc:
Предлагаю убрать файл из репозитория.

alekseirechitskiy replied:
Удалил.

fstk1337 commented on src/assets/images/logo-admin.png:
Не могу разобраться. Показалось, что размеры изображения не соответствуют макету.

fstk1337 commented on src/components/adminPanel/LogIn/Login.module.scss:
```diff
@@ -0,0 +1,46 @@
+
+}
+
+.logIn__title {
+  padding-bottom: 10px;
+  margin-bottom: 40px;
+  border-bottom: 1px solid rgba($color-black, 0.12);
+  font-size: 24px;
+  font-style: normal;
+  font-weight: 400;
+}
```
Верхний отступ заголовка можно побольше. Цвет шрифта, кажется, в макете другой и не указан line-height.

alekseirechitskiy replied:
Исправил все кроме цвета шрифта. Есть предположение, что дизайнеры не досмотрели черный цвет на этой странице 🙂

fstk1337 commented on src/components/adminPanel/LogIn/Login.module.scss:
```diff
@@ -0,0 +1,46 @@
+  margin-bottom: 40px;
+}
+
+.logIn__text {
+  color: $color-black;
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 400;
+  line-height: 24px;
+}
```
Верхний отступ слишком большой, возможно надо в 0.

alekseirechitskiy replied:
Сделал верхний отступ 5 пикселей, так как 0 - маловато.

fstk1337 commented on src/components/adminPanel/ManagePoints/ManagePoints.module.scss:
```diff
@@ -0,0 +1,55 @@
+  }
+}
+
+.managePoints__title {
+  margin: 0;
+  margin-bottom: 64px;
+  margin-right: 24px;
+  font-family: Roboto;
+  font-size: 48px;
+  font-style: normal;
+  font-weight: 600;
+  line-height: 120%;
+  color: $color-black;
+
+  @media (max-width: 1439px) {
+    font-size: 36px;
+    line-height: 110%;
+  }
+
+  @media (max-width: 767px){
+    font-size: 28px;
+    margin-bottom: 30px;
+  }
+}
```
Размер, насыщенность шрифта, line-height и отступ снизу прошу проверить.

alekseirechitskiy replied:
Исправил. Насыщенность шрифта такая же как и в макете.

fstk1337 commented on src/components/adminPanel/ManagePoints/ManagePoints.module.scss:
```diff
@@ -0,0 +1,55 @@
+  }
+}
+
+.managePoints__tabsBox {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
```
В макете есть небольшие внутренние отступы слева и справа, а также внешние отступы у этого элемента.

alekseirechitskiy replied:
Исправил.

fstk1337 commented on src/components/adminPanel/PointsList/PointsList.tsx:
![image](https://github.com/Valeria310/MyGreenApp/assets/45740951/cb5313ce-d636-4c2f-a936-76c182949dbe)

выглядит примерно так у меня в Mozilla Firefox

fstk1337 replied:
Warnings:
1. Warning: validateDOMNesting(...): `<td>` cannot appear as a child of `<a>`.
2. Warning: validateDOMNesting(...): `<a>` cannot appear as a child of `<tbody>`.

alekseirechitskiy replied:
Исправил.

fstk1337 commented on src/components/adminPanel/PointsList/PointsList.tsx:
```diff
@@ -0,0 +1,136 @@
+                    </TableBody>
+                </Table>
+            </TableContainer>
+            <TablePagination
+                slots={{
+                    root: 'div',
+                    toolbar: 'div'
+                }}
+                labelRowsPerPage={'Строк на странице'}
+                labelDisplayedRows={function defaultLabelDisplayedRows({ from, to, count }) {
+                    return `${from}–${to} из ${count !== -1 ? count : `more than ${to}`}`;
+                }}
+                className={classes.pagination}
+                rowsPerPageOptions={[5, 10, 15, { label: 'Все', value: -1 }]}
+                count={tableData.length}
+                rowsPerPage={rowsPerPage}
+                page={page}
+                slotProps={{
+                    select: {
+                        'aria-label': 'rows per page'
+                    },
+                    actions: {
+                        showFirstButton: true,
+                        showLastButton: true
+                    }
+                }}
+                onPageChange={handleChangePage}
+                onRowsPerPageChange={handleChangeRowsPerPage}
+            />
```
Пагинация не совсем по дизайну.

alekseirechitskiy replied:
Дизайн пагинации оговорен с дизайнерами.

fstk1337 commented on src/components/adminPanel/PointsMap/PointsMap.tsx:
```diff
@@ -0,0 +1,122 @@
+    return (
+        <>
+            <Box className={classes.pointsMap__filters}>
+                {filterButtons.map((f) => (
+                    <Chip
+                        className={classes.pointsMap__chip}
+                        key={f.id}
+                        label={f.wasteTitle}
+                        variant={f.isActive ? 'filled' : 'outlined'}
+                        color="primary"
+                        clickable
+                        component="button"
+                        onClick={() => changeButtonStatus(f.id)}
+                    />
+                ))}
```
Отсутствует чипса "Все", высота чипс не по макету. Первая буква заглавная, line-height не соответствует, letter-spacing не задан.

alekseirechitskiy replied:
Исправил.

fstk1337 commented on src/components/adminPanel/ManagePoints/ManagePoints.tsx:
По дизайну Snackbar `Новый пункт приема добавлен успешно` располагается в этом компоненте (либо выше в компоненте страницы).

alekseirechitskiy replied:
Для реализации данной функциональности необходимо общее состояние приложения.
Смогу реализовать эту фичу после внедрения в проект менеджера состояния.

fstk1337 commented on src/components/adminPanel/Point/Point.tsx:
```diff
@@ -0,0 +1,182 @@
+                        >
+                            <Box className={classes.point__tableCol}>Виды вторсырья</Box>
+                            <Box className={classes.point__tableCol}>
+                                {pointData.wasteTypes.map((item) => {
+                                    return (
+                                        <Chip
+                                            key={item}
+                                            className={classes.point__typeChip}
+                                            label={item}
+                                            variant="outlined"
+                                            color="primary"
+                                        />
+                                    );
+                                })}
```
Название чипсы начинается с заглавной буквы.

alekseirechitskiy replied:
Исправил.

fstk1337 commented on src/components/adminPanel/PointForm/PointForm.tsx:
Может быть есть смысл не позволять пользователю вводить больше разрешенного количества символов в каждое поле?

alekseirechitskiy replied:
Поведение компонента следует обсудить с дизайнерами, тестировщиками, бизнес аналитиками.

fstk1337 commented on src/components/adminPanel/PointForm/PointForm.tsx:
```diff
@@ -0,0 +1,671 @@
+                                    Отменить
+                                </Button>
+                            </Box>
+                            <Snackbar
+                                anchorOrigin={{ vertical, horizontal }}
+                                open={open}
+                                autoHideDuration={3000}
+                                onClose={handleClose}
+                                key={vertical + horizontal}
+                            >
+                                <Alert
+                                    onClose={handleClose}
+                                    sx={{
+                                        width: '100%',
+                                        background: '#EDF7ED',
+                                        color: '#1E4620'
+                                    }}
+                                >
+                                    {Object.keys(props).length === 0
+                                        ? 'Новый пункт приема добавлен успешно'
+                                        : 'Пункт приема успешно обновлен'}
+                                </Alert>
+                            </Snackbar>
```
По идее Snackbar должен всплывать на страничке, на которую нас перекидывает после нажатия "Сохранить".

alekseirechitskiy replied:
Для реализации данной функциональности необходимо общее состояние приложения.
Смогу реализовать эту фичу после внедрения в проект менеджера состояния.

fstk1337 commented on src/components/adminPanel/PointForm/PointForm.tsx:
```diff
@@ -0,0 +1,671 @@
+                                >
+                                    Сохранить
+                                </Button>
+                                <Button
+                                    onClick={props.id ? () => navigate(-2) : () => navigate(-1)}
+                                    type="button"
+                                    variant="text"
+                                >
+                                    Отменить
+                                </Button>
```
При отмене редактирования точки логично возвращаться на страницу точки. Сейчас возвращает на общий список точек.

alekseirechitskiy replied:
Исправил.

fstk1337 reviewed (commented):
Отличная работа! Комменты оставил.

alekseirechitskiy committed:
fixed the issues identified by Vitali (
fstk1337)

alekseirechitskiy committed:
Delete .prettierrc

alekseirechitskiy committed:
Delete .editorconfig

alekseirechitskiy committed:
fixed navigation in PointForm, changed styles of the chips in PointsMap

alekseirechitskiy committed:
fixed navigation in PointForm, changed styles of the chips in PointsMap

fstk1337 reviewed (approved):
Славно!

andreikostsen commented:
Блин круто! Леша молодец!

andreikostsen reviewed (approved):
Very good job! Well done!

alekseirechitskiy committed:
added updated MapState.ts;
improved algorithm for drawing checkboxes in PointForm;
improved algorithm of rendering and routing in Point and EditPoint components;
added temporary redirect to the 404 page

alekseirechitskiy committed:
fixed regExp for 'Наименование организации'

alekseirechitskiy committed:
added displaying information to the fields if there is no any data in PointList component

alekseirechitskiy committed:
added improved filters algorithm in PointsMap component

alekseirechitskiy committed:
changed ids in MapState.ts

alekseirechitskiy committed:
Merge branch 'develop' of github.com:Valeria310/MyGreenApp into adminPanel
